tests/integration/test_vulnerability_detector/test_feeds/test_validate_feed_content.py

```diff
@@ -0,0 +1,103 @@
+    {'feed': 'canonical', 'os': 'xenial', 'expected_format': 'application/x-bzip2',
+     'path': '/tmp/com.ubuntu.xenial.cve.oval.xml.bz2', 'extension': 'bz2', 'decompressed_file': '/tmp/xenial.xml',
+     'url':  'https://people.canonical.com/~ubuntu-security/oval/com.ubuntu.xenial.cve.oval.xml.bz2'},
+    {'feed': 'canonical', 'os': 'trusty', 'expected_format': 'application/x-bzip2',
+     'path': '/tmp/com.ubuntu.trusty.cve.oval.xml.bz2', 'extension': 'bz2', 'decompressed_file': '/tmp/trusty.xml',
+     'url':  'https://people.canonical.com/~ubuntu-security/oval/com.ubuntu.trusty.cve.oval.xml.bz2'},
+
+    # Debian
+    {'feed': 'debian', 'os': 'buster', 'expected_format': 'xml', 'path': '/tmp/oval-definitions-buster.xml',
+     'extension': 'xml', 'url':  'https://www.debian.org/security/oval/oval-definitions-buster.xml'},
+    {'feed': 'debian', 'os': 'stretch', 'expected_format': 'xml',  'path': '/tmp/oval-definitions-stretch.xml',
+     'extension': 'xml', 'url':  'https://www.debian.org/security/oval/oval-definitions-stretch.xml'},
+    {'feed': 'debian', 'os': 'jessie', 'expected_format': 'xml',  'path': '/tmp/oval-definitions-jessie.xml',
+     'extension': 'xml', 'url':  'https://www.debian.org/security/oval/oval-definitions-jessie.xml'},
+    {'feed': 'debian', 'os': 'wheezy', 'expected_format': 'xml',  'path': '/tmp/oval-definitions-wheezy.xml',
+     'extension': 'xml', 'url':  'https://www.debian.org/security/oval/oval-definitions-wheezy.xml'}
+]
+
+# NVD
+for year in range(2002, current_year):
+    format_feed_data.append({'feed': 'nvd', 'os': year, 'expected_format': 'application/gzip',
+                             'path': f'/tmp/nvdcve-1.0-{year}.json.gz', 'extension': 'gz',
+                             'url': f'https://nvd.nist.gov/feeds/json/cve/1.0/nvdcve-1.0-{year}.json.gz',
+                             'decompressed_file': f'/tmp/nvd-{year}.json'})
+
+format_feed_data_ids = [f"{item['feed']}_{item['os']}" for item in format_feed_data]
+
+
+@pytest.fixture
+def manage_file(feed, request):
+    """
+    Download and clean test files
+
+    Parameters
+    ----------
+    feed: dict
+        Feed information which comes from an element of 'format_feed_data' list
+    """
+    # Download the file
+    file.download_file(source_url=feed['url'], dest_path=feed['path'])
+
+    # Decompress files
+    if feed['expected_format'] == 'application/gzip':
+        file.decompress_gzip(gzip_file_path=feed['path'], dest_file_path=feed['decompressed_file'])
+    elif feed['expected_format'] == 'application/x-bzip2':
+        file.decompress_bz2(bz2_file_path=feed['path'], dest_file_path=feed['decompressed_file'])
+
+    yield
+
+    # Clean downloaded file/s
+    file.remove_file(file_path=feed['path'])
+
+    if feed['expected_format'] == 'application/gzip' or feed['expected_format'] == 'application/x-bzip2':
+        file.remove_file(file_path=feed['decompressed_file'])
```

Comment:
If none of these cases match the condition, this test will leave lingering files.